components/brave_rewards/resources/android_page/components/contributeBox.tsx

```diff
@@ -185,43 +231,61 @@ class ContributeBox extends React.Component<Props, State> {
     )
   }
 
+  onSettingsToggle = () => {
+    this.setState({ settings: !this.state.settings })
+  }
+
   render () {
     const {
+      enabledMain,
       walletInfo,
       contributionMonthly,
-      enabledMain,
+      enabledContribute,
       reconcileStamp,
       autoContributeList,
-      enabledContribute,
       excludedList,
-      balance
+      balance,
+      ui
     } = this.props.rewardsData
-    const { onlyAnonWallet } = this.props.rewardsData.ui
-    const prefix = this.state.allSitesShown ? 'Hide all' : 'Show all'
     const monthlyList: MonthlyChoice[] = utils.generateContributionMonthly(walletInfo.choices, balance.rates)
     const contributeRows = this.getContributeRows(autoContributeList)
-    const shownRows = this.state.allSitesShown ? contributeRows : contributeRows.slice(0, 5)
+    const excludedRows = this.getExcludedRows(excludedList)
+    const topRows = contributeRows.slice(0, 5)
     const numRows = contributeRows && contributeRows.length
-    const numExcludedRows = excludedList && excludedList.length
+    const numExcludedRows = excludedRows && excludedRows.length
+    const allSites = !(excludedRows.length > 0 || numRows > 5)
+    const { onlyAnonWallet } = ui
 
     return (
       <BoxMobile
         title={getLocale('contributionTitle')}
         type={'contribute'}
         description={getLocale('contributionDesc')}
         toggle={enabledMain}
-        checked={enabledMain && enabledContribute}
+        checked={enabledMain ? enabledContribute : false}
         settingsChild={this.contributeSettings(monthlyList)}
         toggleAction={this.onToggleContribution}
       >
+        {
+          this.state.modalContribute
+          ? <ModalContribute
```

Comment:
was this confirmed with design as this modal is not responsive?

Reply:
I'll enter a new issue for designing a custom mobile design for this modal. It's currently functional/usable on mobile, but could use some tweaks.

Reply:
https://github.com/brave/brave-browser/issues/7613